### PR TITLE
Improvement: complete e2e tests

### PR DESCRIPTION
**Pull request by Zorin95670**
:information_source:  Don't forget to modify the changelog.md before merging this branch.

### CODE REVIEW

File: cypress/e2e/link.feature [hunk 1]
```diff
@@ -0,0 +1,64 @@
+Feature: Test link
```
Comment (justinehell) [inline cypress/e2e/link.feature:1]: `Test link creation`

File: cypress/support/drag-support.js [hunk 2]
```diff
@@ -19,6 +19,9 @@ export function drag(dragSelector, dropSelector) {
     targetCoords = dropSelector;
   }
 
+  console.log(originCoords);
+  console.log(targetCoords);
```
Comment (justinehell) [inline cypress/support/drag-support.js:23]: delete ?

File: cypress/support/step_definitions/html.js [hunk 3]
```diff
@@ -3,9 +3,16 @@ import nunjucks from 'nunjucks';
 
 Then(/^I expect ["'](.*)["'] (?:is|to be) visible$/, (templateSelector) => {
   const selector = nunjucks.renderString(templateSelector, cy.context);
+  console.log(selector);
```
Comment (justinehell) [inline cypress/support/step_definitions/html.js:6]: delete ?

File: cypress/support/step_definitions/html.js [hunk 4]
```diff
@@ -3,9 +3,16 @@ import nunjucks from 'nunjucks';
...
   cy.get(selector).should('be.visible');
 });
 
+Then(/^I expect ["'](.*)["'] (?:is|to be) not visible$/, (templateSelector) => {
+  const selector = nunjucks.renderString(templateSelector, cy.context);
+  console.log(selector);
```
Comment (justinehell) [inline cypress/support/step_definitions/html.js:12]: delete ?

**Review by justinehell — CHANGES_REQUESTED** [review]

File: cypress/e2e/link.feature (hunk 1 above)
Comment (Zorin95670) [reply cypress/e2e/link.feature:1]: done and I rename file in `link-creation.feature`

File: cypress/support/drag-support.js (hunk 2 above)
Comment (Zorin95670) [reply cypress/support/drag-support.js:23]: oups, done

File: cypress/support/step_definitions/html.js (hunk 3 above)
Comment (Zorin95670) [reply cypress/support/step_definitions/html.js:6]: oups, done

File: cypress/support/step_definitions/html.js (hunk 4 above)
Comment (Zorin95670) [reply cypress/support/step_definitions/html.js:12]: oups, done

File: cypress/e2e/link-creation.feature [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  Scenario: Create a invalid link should not make a link appear
+    When I click on "#svg-external1"
+    And I click on "#create-link"
+    And I click on "#svg-server1"
+
+    Then I expect ".link" appear 2 times on screen
+    And I expect "#link-server-laptop_link-server1-external1" is visible
+    And I expect "#link-laptop-network_link-internal1-network1" is visible
+
+  Scenario: Should be able to create multi same link type on different target id
```
Comment (justinehell) [inline cypress/e2e/link-creation.feature:27]: `multi same link type` -> `multiple link with same type` or `multiple links of the same type`

File: cypress/e2e/link-creation.feature [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+    When I click on "#create-link"
+    And I click on "#svg-external1"
+    Then I expect ".link" appear 4 times on screen
+    And I expect "#link-server-laptop_link-server1-external1" is visible
+    And I expect "#link-laptop-network_link-internal1-network1" is visible
+    And I expect "#link-server-laptop_link-server2-internal1" is visible
+    And I expect "#link-server-laptop_link-server2-external1" is visible
+
+  Scenario: Should be able to create multi different link type on different target id
```
Comment (justinehell) [inline cypress/e2e/link-creation.feature:45]: same `multi` -> `multiple` or `multiple links of different types`

**Review by justinehell — CHANGES_REQUESTED** [review]

File: cypress/e2e/delete.feature [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+    Then I expect ".component" appear 5 times on screen
+    And I expect ".component-DefaultModel" appear 3 times on screen
+    And I expect ".component-DefaultContainer" appear 2 times on screen
+    And I expect "#internal1" is visible
+    And I expect "#network1" is visible
+    And I expect "#network2" is visible
+    And I expect "#server1" is visible
+    And I expect "#server2" is visible
+
+  Scenario: Should be able to delete a empty container
```
[the remark this answers is not in the document]
Comment (Zorin95670) [reply cypress/e2e/delete.feature:29]: done

File: cypress/e2e/link-creation.feature (hunk 5 above)
Comment (Zorin95670) [reply cypress/e2e/link-creation.feature:27]: done

File: cypress/e2e/link-creation.feature (hunk 6 above)
Comment (Zorin95670) [reply cypress/e2e/link-creation.feature:45]: done

**Review by justinehell — APPROVED** [review]

**Pull request merged**